src/tests/rules.rs

```diff
@@ -61,3 +61,71 @@ fn no_more_than_5_uses_of_foo_rule() -> Arc<dyn Rule> {
         ]
     }
 }
+
+#[test]
+fn test_rule_options() {
+    RuleTester::run(
+        no_more_than_n_uses_of_foo_rule(),
+        rule_tests! {
+            valid => [
+                {
+                    code => r#"
+                        fn foo() {
+                            let foo = foo;
+                            foo();
+                            foo();
+                        }
+                    "#,
+                    options => 5,
+                }
+            ],
+            invalid => [
+                {
+                    code => r#"
+                        fn foo() {
+                            let foo = foo;
+                            foo();
+                            foo();
+                            foo();
+                        }
+                    "#,
+                    options => 5,
+                    errors => [r#"Can't use 'foo' more than 5 times"#],
+                },
+            ]
+        },
+    );
+}
+
+fn no_more_than_n_uses_of_foo_rule() -> Arc<dyn Rule> {
+    // parse_options => |options| -> usize {
```

Comment:
This is what I was starting to write out before I said "this looks a lot like a manual `Deserialize` implementation" and so instead did the more declarative `options_type`